### PR TITLE
Improve release automation for Helm charts

### PR DESCRIPTION
**Pull request by adriansuarez**
- Get rid of configuration for changelog generator tool and script to perform tagging and updating of versions, which did not do the right thing and dangerously pushed commits and tags to remote.
- Rename scripts/release to scripts/gcp-release to make its purpose clear.
- Add relman.py tool and related files. release/README.md describes how to use this tool to manage release tags and branches, and generate chagnelogs.
- Add package.sh and update-index.sh scripts that package Helm charts in stable/ directory and update the index.yaml file in the gh-pages branch respectively. These scripts can be used to generate artifacts to publish to GitHub releases and to publish the commit to the gh-pages branch that causes the new artifacts to be published.

### CODE REVIEW

**Comment by adriansuarez** [comment]
Pushed another commit that fully automates publishing of GitHub Releases and updating of the GitHub Pages index to include the newly published charts.

File: release/relman.py [hunk 1]
```diff
@@ -0,0 +1,498 @@
+#!/usr/bin/env python
```
Comment (adriansuarez) [inline release/relman.py:1]: This is an existing tool that was already being used in the `nuodb-control-plane` repo.

File: scripts/package.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# Change to root directory
+cd "$(dirname "$0")/.."
+
+# Create directory for packaged Helm charts
+dest="package/v$(./get-version.sh)"
+mkdir -p "$dest"
+
+# Package all Helm charts
+charts="$(find stable -maxdepth 2 -name Chart.yaml -exec dirname {}  \;)"
```
Comment (sivanov-nuodb) [inline scripts/package.sh:18]: Can we have the repository as an input parameter? We may want to consider adding the `incubator` repo in the future.

File: scripts/update-index.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+git merge --ff-only origin/gh-pages
+
+# Update index with new Helm charts
+: ${GH_RELEASES_URL:="https://github.com/nuodb/nuodb-helm-charts/releases/download"}
+helm repo index package --merge index.yaml --url "$GH_RELEASES_URL"
+
+# Commit and push change if PUSH_UPDATE=true
+if [ "$PUSH_UPDATE" = true ]; then
+    mv package/index.yaml .
+    git add index.yaml
```
Comment (sivanov-nuodb) [inline scripts/update-index.sh:29]: Having the `index.yaml` in a sub-directory named after the repository name will allow us to support several Helm repos.

**Review by sivanov-nuodb — APPROVED** [review]
Looks good.
I've added some suggestions about supporting multiple Helm repositories. E.g. `stable` and `incubator`.

**Review by bkelly-ndb — APPROVED** [review]

File: release/relman.py
[the remark this answers is not in the document]
Comment (adriansuarez) [reply release/relman.py:1]: My eventual plan is to make this a submodule, but for now, I will just copy it over.

File: README.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ Helm 2 has reached it's end of life. The last Helm 2 security patch was released
 To add the NuoDB charts to your local helm repository, run:
 
 ```
-helm repo add nuodb https://storage.googleapis.com/nuodb-charts
+helm repo add nuodb https://nuodb.github.io/nuodb-helm-charts
```
[the remark this answers is not in the document]
Comment (adriansuarez) [reply README.md:34]: Thanks for catching that. `grep -R` ignores hidden files/directories.

File: .github/workflows/publish-charts.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+          fi
+          if [ ! -f "changelogs/v${VERSION}.md" ]; then
+            echo "No changelog found for $VERSION" 2>&1
+            exit 1
+          fi
+          echo "tag=v$VERSION" >> $GITHUB_OUTPUT
+      - name: Package Helm charts
+        run: ./scripts/package.sh
+      - name: Publish GitHub release
+        run: gh release create ${{ steps.vars.outputs.tag }} -F changelogs/${{ steps.vars.outputs.tag }}.md package/${{ steps.vars.outputs.tag }}/*
```
[the remark this answers is not in the document]
Comment (adriansuarez) [reply .github/workflows/publish-charts.yml]: You can just delete the release and let the trigger recreate it. I'm not sure what happens when it already exists, but I can test it.

[the remark this answers is not in the document]
Comment (adriansuarez) [reply .github/workflows/publish-charts.yml]: The workflow run will fail with `a release with the same tag name already exists: <tag>`.

See the following run: https://github.com/adriansuarez/nuodb-helm-charts/actions/runs/5999941437/job/16271055645

[the remark this answers is not in the document]
Comment (adriansuarez) [reply .github/workflows/publish-charts.yml]: Deleting the release and re-running the workflow was simple enough: https://github.com/adriansuarez/nuodb-helm-charts/releases

[the remark this answers is not in the document]
Comment (sivanov-nuodb) [reply .github/workflows/publish-charts.yml]: Thanks for testing it out!

**Pull request merged**